src/Orleans.CodeGenerator/Generators/SerializerGenerator.cs

```diff
@@ -346,11 +345,10 @@ private MemberDeclarationSyntax GenerateSerializerMethod(INamedTypeSymbol type,
         private MemberDeclarationSyntax GenerateDeepCopierMethod(INamedTypeSymbol type, List<FieldInfoMember> fields, SemanticModel model)
         {
             var originalVariable = IdentifierName("original");
-            var inputVariable = IdentifierName("input");
-            var resultVariable = IdentifierName("result");
 
             var body = new List<StatementSyntax>();
-            if (type.HasInterface(wellKnownTypes.ImmutableAttribute))
+            if (type.HasAttribute(wellKnownTypes.ImmutableAttribute)
+                || SymbolEqualityComparer.Default.Equals(wellKnownTypes.Immutable_1, type))
```

Comment:
Does SymbolEqualityComparer.Default.Equals return true for comparisons between `Immutable<>` and `Immutable<MyClass>`?

Reply:
This is just a special case for the `Immutable<T>` type itself that gets generated for `Orleans.Core`. Added it just in case that ends up actually used for deep copying (though it shouldn't).

Reply:
Maybe it would be better to actually skip emitting that `Immutable<T>` serializer altogether as it should always us the built-in serializer?

Reply:
We should consider adding a `[SkipCodeGeneration]` attribute, or something more specific and add it to `Immutable<T>`